extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -236,7 +240,9 @@ static boolean isSameOrigin(HttpServerRequest request, String origin) {
                         request.absoluteURI(), origin);
                 return false;
             }
-            if (substringMatch(origin, request.scheme().length() + 3, request.host(), true)) {
+            var authority = request.authority();
+            if (authority != null && authority.host() != null
```

Comment:
I guess it would be better to cover the move from `.host()` to `.authority` in a Vert.x 5 upgrade as I guess more code around uses it, there should be some tests confirming both are equivalent... Here it looks like it is not guaranteed to identity the same origin as two extra checks for null are added, may be we should not check for null at all as the host must be set, but I'd just keep `request.host()` for now